sector-builder/src/scheduler.rs

```diff
@@ -267,9 +273,39 @@ impl<T: KeyValueStore, S: SectorStore> SectorMetadataManager<T, S> {
     }
 
     // Produces a vector containing metadata for all sealed sectors that this
-    // SectorBuilder knows about.
-    pub fn get_sealed_sectors(&self) -> Result<Vec<SealedSectorMetadata>> {
-        Ok(self.state.sealed.sectors.values().cloned().collect())
+    // SectorBuilder knows about. Includes sector health-information on request.
+    pub fn get_sealed_sectors(&self, check_health: bool) -> Result<Vec<GetSealedSectorResult>> {
+        use rayon::prelude::*;
+
+        let sectors_iter = self.state.sealed.sectors.values().cloned();
+
+        if !check_health {
+            return Ok(sectors_iter
+                .map(GetSealedSectorResult::WithoutHealth)
+                .collect());
+        }
+
+        let with_path: Vec<(PathBuf, SealedSectorMetadata)> = sectors_iter
+            .map(|meta| {
+                let pbuf = self
+                    .sector_store
+                    .manager()
+                    .sealed_sector_path(&meta.sector_access);
+
+                (pbuf, meta)
+            })
+            .collect();
+
+        // compute sector health in parallel using workers from rayon global
+        // thread pool
+        with_path
+            .par_iter()
+            .cloned()
```

Comment:
you should be able to call into_par_iter and avoid the clone

Reply:
good call. done.